snapcraft/project/_sanity_checks.py

```diff
@@ -32,7 +35,21 @@
     re.compile(r"^gui(/.*\.(png|svg|desktop))?$"),
 }
 
-# TODO: Add conduct_environment_sanity_check() to be run in the build environment
+
+def conduct_build_environment_sanity_check(provider: str):
+    if provider == "multipass":
+        _check_multipass_installed()
+
+
+def _check_multipass_installed():
+    if not shutil.which("multipass"):
```

Comment:
how about
```python
if shutil.which("multipass"):
    return

if sys.platform != "linux":
    raise errors.MultipassMissingNonLinuxError()

if shutil.which("snap"):
   raise errors.MultipassMissingLinuxError()
else:
    raise errors.SnapMissingLinuxError()
```

Which makes it a little less nested and easier to add future checks in as well.